velox/expression/CastExpr-inl.h

```diff
@@ -48,6 +51,67 @@ inline std::exception_ptr makeBadCastException(
       false));
 };
 
+/// @brief Convert the unscaled value of a decimal to varchar and write to raw
+/// string buffer from start position.
+/// @tparam T The type of input value.
+/// @param unscaledValue The input unscaled value.
+/// @param scale The scale of decimal.
+/// @param maxVarcharSize The estimated max size of a varchar.
+/// @param startPosition The start position to write from.
+/// @return A string view.
+template <typename T>
+StringView convertToStringView(
+    T unscaledValue,
+    int32_t scale,
+    int32_t maxVarcharSize,
+    char*& startPosition) {
+  char* initialPosition = startPosition;
+  if (unscaledValue == 0) {
+    *startPosition++ = '0';
+  } else {
+    if (unscaledValue < 0) {
+      *startPosition++ = '-';
+      unscaledValue = -unscaledValue;
+    }
+    auto [position, errorCode] = std::to_chars(
+        startPosition,
+        startPosition + maxVarcharSize,
+        unscaledValue / DecimalUtil::kPowersOfTen[scale]);
+    VELOX_DCHECK_EQ(
+        errorCode,
+        std::errc(),
+        "Failed to cast decimal to varchar: {}",
+        std::make_error_code(errorCode).message());
+    startPosition = position;
+
+    if (scale > 0) {
+      *startPosition++ = '.';
+      uint128_t fraction = unscaledValue % DecimalUtil::kPowersOfTen[scale];
+      // Append leading zeros.
+      int numLeadingZeros =
+          std::max(scale - fmt::detail::count_digits(fraction), 0);
+      std::memset(startPosition, '0', numLeadingZeros);
+      startPosition += numLeadingZeros;
+      // Append remaining fraction digits.
+      auto [position, errorCode] = std::to_chars(
+          startPosition, startPosition + maxVarcharSize, fraction);
+      VELOX_DCHECK_EQ(
+          errorCode,
+          std::errc(),
+          "Failed to cast decimal to varchar: {}",
+          std::make_error_code(errorCode).message());
+      startPosition = position;
+    }
+  }
+  int32_t len = startPosition - initialPosition;
+  auto stringView = StringView(initialPosition, len);
+  if (StringView::isInline(len)) {
```

Comment:
nit: I suggest to keep this function "simple" and do only one thing, convert the value to string, and move this logic to the caller.

Reply:
Fixed.